src/main/java/com/winner_is_kungen/tda367/model/Component.java

```diff
@@ -84,8 +84,8 @@ public int getNrOutputs() {
 	 * @param outChannel A Integer specifying which input is used.
 	 */
 	void addListener(ComponentListener listener, int inChannel, int outChannel) {
+		listener.update(UUID.randomUUID().toString(),logic(inputChannels)[outChannel],inChannel);
```

Comment:
The old way (that you change back to here) is probably worse. If we have a self-calling configuration this will update all the channels' values but only notify the newly connected listener of the new value, causing all the others to be out of sync.